action.sh

```diff
@@ -293,7 +293,12 @@ function start_vm {
     ${maintenance_policy_flag} \
     --labels=gh_ready=0,vm_id=${VM_ID} \
     --metadata=startup-script="$startup_script"
- 
+
+  if [ $? -ne 0 ]; then
```

Comment:
I'm surprised that this script isn't already exiting this way, since safety_on is called prior to the enclosing function (start_vm).